### DISCUSSION TITLE
Allure Report extension for Azure DevOps CORS error

### DISCUSSION THREAD

**Discussion by vvgorceac** [allure-framework/allure2#2338] [Questions & Support] [answered]
Hi, we recently installed [Allure extension for Azure](https://marketplace.visualstudio.com/items?itemName=qameta.allure-azure-pipelines) on our Azure DevOps

After pipeline has been executed and report was generated, I'm trying to reach AllureReport tab, however it is loading infinitely 

![image](https://github.com/allure-framework/allure2/assets/32510523/88274bfb-eb16-4d97-b0e9-5fa06c55aa30)

CORS Error in console: 

`
index.html:1 Access to fetch at 'https://dev.azure.com/PROJECT_NAME/AN_UUID/_apis/build/builds/2597/attachments/allure.report?api-version=7.1-preview.2' from origin 'https://ACTUAL.DEVOPS.DOMAIN' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource. If an opaque response serves your needs, set the request's mode to 'no-cors' to fetch the resource with CORS disabled.
`

`
index.js:1  
       GET https://dev.azure.com/PROJECT_NAME/AN_UUID/_apis/build/builds/2597/attachments/allure.report?api-version=7.1-preview.2 net::ERR_FAILED`

Our devops is hosted on a custom domain https://ACTUAL.DEVOPS.DOMAIN, but by some reason extension is trying to reach another domain https://dev.azure.com. 

Currently I am trying to figure out is it an issue with the extension or DevOps configuration, any help would be much appreciated.

**Comment by epszaw** [comment]
Hey, @vvgorceac! I published new version (1.0.4) which uses client from official azure SDK. Could you please try to open a fresh version?

**Comment by vvgorceac** [comment]
Hi @epszaw, 
many thanks for providing updated version, it helped to solve a CORS issue, however now there is another problem. 
Navigating to Allure Report tab will end up with infinite loading screen again, reason for this will be failed request:
` GET https://custom.devops.domain/PROJECT/UUID/_apis/build/builds/2610/attachments/allure.report `
Response code is 400 
Body contains following message (it is translated from a different language, but should be clear): 
**The requested REST API version (7.2) is out of range for this server. The latest REST API version supported by this server is version 7.0.**

I tried to open https://custom.devops.domain/PROJECT/UUID/_apis/build/builds/2610/attachments/allure.report in a new browser tab and got a JSON response which contains link to the HTML file, opening this link leads to downloading the correct html report. 

Also checked in postman, works well with version set to 7.0
❌ application/json;api-version=**7.2-preview.2**;excludeUrls=true;enumsAsNumbers=true;msDateFormat=true;noArrayWrap=true
✅  application/json;api-version=**7.0**;excludeUrls=true;enumsAsNumbers=true;msDateFormat=true;noArrayWrap=true

Is there a way to select/define needed API version in extension settings to avoid this issue ?

**Reply by epszaw** [reply, marked as answer]
The best solution would be to update your server version to make it compatible with the official azure SDK. Maybe, it's just a stub about the incompatibility. So, I published a new version (1.0.5), where attachments request performs without using the SDK. Could you please try the version?

**Comment by vvgorceac** [comment]
Many thanks @epszaw, issue is solved :)

**Reply by epszaw** [reply]
Glad to hear that! :)
